compiler/src/dotty/tools/dotc/printing/PlainPrinter.scala

```diff
@@ -177,6 +177,8 @@ class PlainPrinter(_ctx: Context) extends Printer {
             keywordStr(" match ") ~ "{" ~ casesText ~ "}" ~
             (" <: " ~ toText(bound) provided !bound.isAny)
         }.close
+      case tp: PreviousErrorType if ctx.settings.XprintTypes.value =>
+        "<error>" // do not print previously reported error message because they may try to print this error type again recuresevely
       case tp: ErrorType =>
         s"<error ${tp.msg.rawMessage}>"
```

Comment:
The infinite recursion happens here. The message may print a tree that contains this error type again.

Reply:
PlainPrinter already has some logic to avoid infinite loops with `limiter` and `maxToTextRecursions`, it'd be good to see if we can extend that to also account for the loops here